docs/netconfig.md

```diff
@@ -0,0 +1,106 @@
+# Network and HTTP configuration
+
+Git Credential Manager Core's network and HTTP(S) behavior can be configured in a few different ways via [environment variables](environment.md) and [configuration options](configuration.md).
+
+## HTTP Proxy
+
+If your computer sits behind a network firewall that requires the use of a proxy server to reach repository remotes or the wider Internet, there are various methods for configuring GCM to use a proxy.
+
+The simplist way to configure a proxy for _all_ HTTP(S) remotes is to [use the standard Git HTTP(S) proxy setting `http.proxy`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-httpproxy).
+
+For example to configure a proxy for all remotes for the current user:
+
+```shell
+git config --global http.proxy http://proxy.example.com
+```
+
+To specify a proxy for a particular remote you can [use the `remote.<name>.proxy` repository-level setting](https://git-scm.com/docs/git-config#Documentation/git-config.txt-remoteltnamegtproxy), for example:
+
+```shell
+git config --local remote.origin.proxy http://proxy.example.com
+```
+
+The advantage to using these standard configuration options is that in addition to GCM being configured to use the proxy, Git itself will be configured at the same time. This is probably the most commonly desired case in environments behind an Internet-blocking firewall.
+
+### Authenticated proxies
+
+Some proxy servers do not accept anonymous connections and require authentication. In order to specify the credentials to be used with a proxy, you can specify the username and password as part of the proxy URL setting.
+
+The format follows [RFC 3986 section 3.2.1](https://tools.ietf.org/html/rfc3986#section-3.2.1) by including the credentials in the 'user information' part of the URI. The password is optional.
+
+```text
+protocol://username[:password]@hostname
+```
+
+For example, to specify the username `john.doe` and the password `letmein123` for the proxy server `proxy.example.com`:
+
+```text
+https://john.doe:letmein123@proxy.example.com
+```
+
+If you have special characters (as defined by [RFC 3986 section 2.2](https://tools.ietf.org/html/rfc3986#section-2.2)) in your username or password such as `:`, `@`, or any other non-URL friendly character you can URL-encode them ([section 2.1](https://tools.ietf.org/html/rfc3986#section-2.2)).
+
+For example, a space character would be encoded with `%20`.
+
+### Other proxy options
+
+GCM Core supports other ways of configuring a proxy for convenience and compatibility.
+
+1. GCM-specific configuration options (_**only** respected by GCM_):
+   - `credential.httpProxy`
```

Comment:
is it worth mentioning this is being deprecated here too?

Reply:
The line before this says:

> ..for convenience and compatibility.

Perhaps I should make it even more explicit still?

Reply:
I think it doesn't hurt to just have (deprecated) in the notes